compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/CallAndReferenceGenerator.kt

```diff
@@ -183,6 +185,7 @@ class CallAndReferenceGenerator(
         return coneType.toIrType() as IrSimpleType
     }
 
+    // TODO: refactor to share some logic with suspend conversion on arguments (maybe introduce AdapterGenerator?)
```

Comment:
@mglukhikh If splitting all adapter-related code into `AdapterGenerator` sounds reasonable, I can do that in a follow-up PR.

Reply:
> If splitting all adapter-related code into AdapterGenerator sounds reasonable, I can do that in a follow-up PR.

Yes, it sounds good.

Reply:
Done at the 3rd commit, which is mostly mechanical migration, plus a couple visibility adjustments and small function/check renaming for consistency.